### PR TITLE
i18n support, with traduora

### PR DESCRIPTION
**Pull request by TerryGeng**
Implements #172.

Thank @azlux for hosting traduora 👍 👍 👍 .

- [x] i18n support for the _cli_ of the bot.
- [x] i18n support for the web interface.

How things work:

1. All strings are now stored in the `lang/` folder.

2. The `sync_translation.py` script will fetch the available translation from @azlux's server and store them into the `lang/` folder.

3. When starting the bot, it will first try to use the language specified by `--lang` argument. If not provided, it will look for the `language` in `[bot]` section. The default value is `en_US`.

4. In the future, all changes in the strings of the bot should be done in the `en_US` file. The `sync_translation.py` script will automatically import it to the server, so new strings are open for translating. This upload will only happen if writable API keys are found in environment variables (set by CI).

Problem:
Do we have to upload all language files? I was thinking about this idea, but I'm afraid some people will modify a string in one language and leave that string in other languages untouched so the entire thing is messed up. Is there any solution to this?

### CODE REVIEW

File: constants.py [hunk 1]
```diff
@@ -1,24 +1,54 @@
...
+            string = default_lang_dict['cli'][option]
+    except KeyError:
+        raise KeyError("Missed strings in language file: '{string}'. ".format(string=option))
+    return _tr(string, *argv, **kwargs)
+
 
-def strings(option, *argv, **kwargs):
+def tr_web(option, *argv, **kwargs):
     try:
-        string = var.config.get("strings", option)
```
Comment (azlux) [inline constants.py:6]: You should keep it for "user can override a string" purpose :
- New PR can just add one string into the config until a translator guy add a new term into traduora
- People can still change some strings for fun.

File: lang/sync_translation.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+
+import os
+import requests
+
+base_url = "https://translate.azlux.fr/api/v1"
+project_id = "4aafb197-3282-47b3-a197-0ca870cf6ab2"
+r_client = "be8215d4-2417-49db-9355-c418f26dc3f4"
+r_secret = "MIMvdnECLkmTZyCQT4DekONN53EOSsj3"
```
Comment (azlux) [inline lang/sync_translation.py:9]: Should we keep the client/secret public ?

File: lang/sync_translation.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        lang_code = translation['locale']['code']
+        print(f" - Fetching {lang_code}")
+        params = {'locale': lang_code,
+                  'format': 'jsonnested'}
+        r = requests.get(f"{base_url}/projects/{project_id}/exports", params=params, headers=headers)
+        with open(lang_code, "wb") as f:
+            f.write(r.content)
+
+
+def push_strings():
```
Comment (azlux) [inline lang/sync_translation.py:38]: Not sure about this one. you delegate the strings creating to this script.
I'm not reassured of the impact of re-import everything everytime.
Are other have opinions ?

File: .drone.yml [hunk 4]
```diff
@@ -61,6 +59,7 @@ steps:
       - echo $version > /mnt/botamusique/version
       - sed -i "s/version = 'git'/version = '$version'/" mumbleBot.py
       - curl --silent "https://api.github.com/repos/azlux/botamusique/releases/latest" | jq -r '.body' | pandoc --from gfm --to html - --output - > /mnt/botamusique/changelog
+      - (cd lang && ./sync_translation.py)
       - rm -rf .git*
```
Comment (azlux) [inline .drone.yml:63]: with the `cd` you are in the bad folder to the next step.

**Comment by TerryGeng** [comment]
So after some discussion, @azlux and I believe that if we still store some user-defined string in the mini file, the logic will be messy. So we decided from now on, users should directly modify their translation file to redefine strings.

As for syncing strings between traduora and our repo, we will fetch all available translation files from traduora and push them into our repo through a cron job. Meanwhile, every time a commit is added to the master branch, the CI will be triggered to upload all translation files into the traduora server, so people can directly modify translation through files.

File: .drone.yml (hunk 4 above)
Comment (TerryGeng) [reply .drone.yml]: No. If I cd in `()`, it won't affect the next line

**Pull request merged**